### PR TITLE
feat: implement multicall

### PR DESCRIPTION
**Pull request by nchamo**
We now implemented `Multicall` and added some tests that use it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "postpublish": "pinst --enable"
   },
   "dependencies": {
-    "@mean-finance/dca-v2-core": "^1.0.0-rc.8"
+    "@mean-finance/dca-v2-core": "^1.0.0-rc.9"
```
Comment (nchamo) [inline package.json:59]: Needed to update to the latest version that doesn't have the bug

**Review by alejoamiras — APPROVED** [review]
This is awesome ! Great work ... One Q: Would you endulge me with one "attack test" ? Trying to permit / modify / withdraw something thats not owned by the user ? 
Better to be safe than encounter some weird loophole ? 🥺

File: deploy/000_register_hub.ts [hunk 2]
```diff
@@ -28,7 +29,7 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
     const chainlinkDeployment = await hre.deployments.deploy('Oracle', {
       from: deployer,
       contract: { abi: CHAINLINK_ABI, bytecode: CHAINLINK_BYTECODE },
-      args: [WETH, CHAINLINK_REGISTRY, governor],
+      args: [WETH, CHAINLINK_REGISTRY, BigNumber.from(2).pow(32).sub(1), governor],
```
Comment (alejoamiras) [inline deploy/000_register_hub.ts:32]: also, this is due to a dca-v2-core change, right ?

Comment (nchamo) [reply deploy/000_register_hub.ts:32]: Indeed

**Pull request merged**